DOCSP-43368: Cleanup (#66)
mongodb/docs-c: source/install-from-source.txt

.. _c-install-from-source:

===========================================
Building the C Driver Libraries from Source
===========================================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol
   
This page details how to download, unpack, configure, and build ``libbson`` and

> ``libmongoc`` from their original source-code form.

Author: This page is not standardized, just preserved from the old docs, but happy to make these edits

.. tip:: Extra information

   Dropdowns (like this one) contain extra information and explanatory details
   that are not required to complete the tutorial, but may be helpful for curious
   readers, and more advanced users that want an explanation of the meaning of
   certain tutorial steps.

> The following page uses a few named "variables" that you must decide up-front.

Author: I think this refers to $VERSION, $SOURCE, etc

.. tabs::

   .. tab:: Using ``wget`` + ``tar``
      :tabid: wget-tar

      .. code-block:: none
        
        ## Download using wget:
        $ wget "https://github.com/mongodb/mongo-c-driver/archive/refs/tags/$VERSION.tar.gz" \
             --output-document="mongo-c-driver-$VERSION.tar.gz"
        ## Extract using tar:
        $ tar xf "mongo-c-driver-$VERSION.tar.gz"

   .. tab:: Using ``curl`` + ``tar``
      :tabid: curl-tar

      .. code-block:: none
        
        ## Using curl:
        $ curl "https://github.com/mongodb/mongo-c-driver/archive/refs/tags/$VERSION.tar.gz" \
             --output="mongo-c-driver-$VERSION.tar.gz"
        ## Extract using tar:
        $ tar xf "mongo-c-driver-$VERSION.tar.gz"

   .. tab:: PowerShell
      :tabid: powershell

      .. code-block:: none

        ## Use Invoke-WebRequest:
        PS> $url = "https://github.com/mongodb/mongo-c-driver/archive/refs/tags/$VERSION.zip"
        PS> $file = "mongo-c-driver-$VERSION.zip"
        PS> Invoke-WebRequest -UseBasicParsing -Uri $url -OutFile $file
        ## Extract using Expand-Archive:
        PS> Expand-Archive mongo-c-driver-$VERSION.zip

> The above commands will create a new directory ``mongo-c-driver-$VERSION`` within
> the directory in which you ran the ``tar`` / ``Expand-Archive`` command (**note**:
> PowerShell will create an additional intermediate subdirectory of the same
> name). This directory is the root of the driver source tree (which we refer to
> as ``$SOURCE`` in these documents). The ``$SOURCE`` directory should contain the
> top-level ``CMakeLists.txt`` file.

Reviewer: S: Since there are a couple of decision paths to represent here (tar / expand-archive / powershell), I would probably put a different block of text beneath each tab. (You can also use `include` statements to avoid repetition.) Maybe something like this for PowerShell, with some style guide edits included:

```suggestion
The preceding commands create the ``[intermediate directory name]`` directory 
 within the directory in which you ran them. The ``[intermediate directory]`` directory
 contains the ``mongo-c-driver-$VERSION`` directory, which is the root of the driver
 source tree. This documentation refers to this directory as ``$SOURCE``. 
 The ``$SOURCE`` directory contains the top-level ``CMakeLists.txt`` file.
```

Obtaining Prerequisites
-----------------------

> In order to build the project, a few prerequisites need to be available.
> 
> Both ``libmongoc`` and ``libbson`` projects use `CMake <https://cmake.org>`__ for build configuration.

Reviewer: S:

```suggestion
You must install `CMake <https://cmake.org>`__ to configure the ``libmongoc`` and
``libbson`` projects.
```

> .. note::
> 
>    It is *highly recommended* -- but not *required* -- that you download the
>    latest stable CMake available for your platform.

Reviewer: S: remove from note and append to prev. paragraph

```suggestion
We highly recommended that you download the latest stable version of
CMake that is available for your platform.
```

.. tip:: Getting the Latest CMake
   
   A new stable release of CMake can be obtained from
   `the CMake downloads page <https://cmake.org/download/#latest>`__.

>    For Windows and macOS, simply download the CMake ``.msi`` / ``.dmg`` (not the

Reviewer: How about different tabs for the OSes, similar to what you did before?

>    For Windows and macOS, simply download the CMake ``.msi`` / ``.dmg`` (not the

Reviewer: I: [slashes](https://www.mongodb.com/docs/meta/style-guide/style/punctuation/#slashes)

```suggestion
   For Windows and macOS, simply download the CMake ``.msi`` or ``.dmg`` (not the
```

(using tabs would also solve this issue)

   ``.zip`` / ``.tar.gz``) and use it to install CMake.

   On Linux, download the self-extracting shell script (ending with ``.sh``) and
   execute it using the ``sh`` utility, passing the appropriate arguments to
   perform the install. For example, with the CMake 3.27.0 on the ``x86_64``
   platform, the following command can be used on the
   ``cmake-3.27.0-linux-x86_64.sh`` script:

   .. code-block:: none

      $ sh cmake-3.27.0-linux-x86_64.sh --prefix="$HOME/.local" --exclude-subdir --skip-license

   Assuming that ``$HOME/.local/bin`` is on your ``$PATH`` list, the ``cmake`` command
   for 3.27.0 will then become available.
   

>    The ``--help`` option can be passed to the shell script for more information.

Reviewer: S: remove from tip box; seems like a pretty important step to install it

> For the remainder of this page, it will be assumed that ``cmake`` is available as

Reviewer: I: I'm not as anti-passive-voice as the [style guide](https://www.mongodb.com/docs/meta/style-guide/writing/use-active-voice/), but it seems easy to avoid here

```suggestion
This page assumes that ``cmake`` is available as
```

   :depth: 2
   :class: singlecol
   
This page details how to download, unpack, configure, and build the ``libbson`` and
``libmongoc`` libraries from their source code.

.. tip:: Extra information

   Admonitions (like this one) contain extra information and explanatory details
   that are not required to complete the tutorial, but may be helpful for curious

>    readersx and more advanced users that want an explanation of the meaning of

Reviewer: VIM? 😠 

```suggestion
   readers and more advanced users that want an explanation of the meaning of
```

Author: oops
